runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -58,193 +54,137 @@
-            new Function<byte[], WindowedValue<T>>() {
-          @Override
-          public WindowedValue<T> apply(byte[] bytes) {
-            return CoderHelpers.fromByteArray(bytes, windowedValueCoder);
-          }
-        });
-      }
-      return windowedValues;
-    }
+  public EvaluationContext(JavaSparkContext jsc, Pipeline pipeline,
+                           JavaStreamingContext jssc, long timeout) {
+    this(jsc, pipeline);
+    this.jssc = jssc;
+    this.timeout = timeout;
+    this.state = State.RUNNING;
   }
 
-  protected JavaSparkContext getSparkContext() {
+  JavaSparkContext getSparkContext() {
     return jsc;
   }
 
+  public JavaStreamingContext getStreamingContext() {
+    return jssc;
+  }
+
   public Pipeline getPipeline() {
     return pipeline;
   }
 
-  protected SparkRuntimeContext getRuntimeContext() {
+  public SparkRuntimeContext getRuntimeContext() {
     return runtime;
   }
 
   public void setCurrentTransform(AppliedPTransform<?, ?, ?> transform) {
     this.currentTransform = transform;
   }
 
-  protected AppliedPTransform<?, ?, ?> getCurrentTransform() {
-    return currentTransform;
-  }
-
-  protected <T extends PInput> T getInput(PTransform<T, ?> transform) {
+  public <T extends PInput> T getInput(PTransform<T, ?> transform) {
     checkArgument(currentTransform != null && currentTransform.getTransform() == transform,
         "can only be called with current transform");
     @SuppressWarnings("unchecked")
     T input = (T) currentTransform.getInput();
     return input;
   }
 
-  protected <T extends POutput> T getOutput(PTransform<?, T> transform) {
+  public <T extends POutput> T getOutput(PTransform<?, T> transform) {
     checkArgument(currentTransform != null && currentTransform.getTransform() == transform,
         "can only be called with current transform");
     @SuppressWarnings("unchecked")
     T output = (T) currentTransform.getOutput();
     return output;
   }
 
-  protected  <T> void setOutputRDD(PTransform<?, ?> transform,
-      JavaRDDLike<WindowedValue<T>, ?> rdd) {
-    setRDD((PValue) getOutput(transform), rdd);
+  public void putDataset(PTransform<?, ?> transform, Dataset dataset) {
+    putDataset((PValue) getOutput(transform), dataset);
+  }
+
+  public void putDataset(PValue pvalue, Dataset dataset) {
+    try {
+      dataset.setName(pvalue.getName());
+    } catch (IllegalStateException e) {
+      // name not set, ignore
+    }
+    datasets.put(pvalue, dataset);
+    leaves.add(dataset);
   }
 
-  protected  <T> void setOutputRDDFromValues(PTransform<?, ?> transform, Iterable<T> values,
-      Coder<T> coder) {
-    pcollections.put((PValue) getOutput(transform), new RDDHolder<>(values, coder));
+  <T> void putBoundedDatasetFromValues(PTransform<?, ?> transform, Iterable<T> values,
+                                       Coder<T> coder) {
+    datasets.put((PValue) getOutput(transform), new BoundedDataset<>(values, jsc, coder));
```

Comment:
This is deliberate. Previous version used ```pcollections``` directly. Unless you mean this was a mistake in the original version?